File: backend/src/main/java/com/bytechat/services/ChannelService.java
package com.bytechat.services;

import com.bytechat.dto.response.ChannelResponse;
import com.bytechat.dto.response.UserResponse;
import com.bytechat.entity.Channel;
import com.bytechat.entity.User;

import java.util.List;

public interface ChannelService {
    ChannelResponse createChannel(Long workspaceId, String name, String description, boolean isPrivate, boolean isDefault, User currentUser);
    List<ChannelResponse> getWorkspaceChannels(Long workspaceId, User currentUser);
    List<ChannelResponse> getArchivedChannels(Long workspaceId, User currentUser);
    List<ChannelResponse> getDeletedChannels(Long workspaceId, User currentUser);
    Channel getChannel(Long channelId);
    void addMember(Long channelId, User user);
    List<UserResponse> getChannelMembers(Long channelId);
    void inviteUser(Long channelId, String email, User currentUser);
    void acceptInvite(Long notificationId, User currentUser);
    void archiveChannel(Long channelId, User currentUser);
    void leaveChannel(Long channelId, User currentUser);
    void deleteChannel(Long channelId, User currentUser); // Now soft delete
    void permanentlyDeleteChannel(Long channelId, User currentUser);
    void restoreChannel(Long channelId, User currentUser);
    void transferOwnership(Long channelId, Long newOwnerId, User currentUser);
}

File: frontend/src/services/channelService.js
import api from './api';

const channelService = {
  createChannel: (workspaceId, name, description, isPrivate) => 
    api.post(`/channels/workspace/${workspaceId}`, null, {
      params: { name, description, isPrivate }
    }),
  getWorkspaceChannels: (workspaceId) => api.get(`/channels/workspace/${workspaceId}`),
  getArchivedChannels: (workspaceId) => api.get(`/channels/workspace/${workspaceId}/archived`),
  getDeletedChannels: (workspaceId) => api.get(`/channels/workspace/${workspaceId}/deleted`),
  inviteUser: (channelId, email) => api.post(`/channels/${channelId}/invite`, { email }),
  leaveChannel: (channelId) => api.post(`/channels/${channelId}/leave`),
  deleteChannel: (channelId) => api.delete(`/channels/${channelId}`),
  permanentlyDeleteChannel: (channelId) => api.delete(`/channels/${channelId}/permanent`),
  restoreChannel: (channelId) => api.post(`/channels/${channelId}/restore`),
  archiveChannel: (channelId) => api.post(`/channels/${channelId}/archive`),
  transferOwnership: (channelId, newOwnerId) => api.post(`/channels/${channelId}/transfer-ownership`, null, {
    params: { newOwnerId }
  }),
};

export default channelService;
